cairo_zero/kakarot/precompiles/kakarot_precompiles.cairo

```diff
@@ -78,44 +72,63 @@ namespace KakarotPrecompiles {
-                // Deploy account -
-                // order of returned values in memory matches the explicit ones in the other branch
-                Starknet.deploy(caller_address);
-            } else {
-                tempvar syscall_ptr = syscall_ptr;
-                tempvar pedersen_ptr = pedersen_ptr;
-                tempvar range_check_ptr = range_check_ptr;
-                tempvar caller_starknet_address = caller_starknet_address;
-            }
-            let syscall_ptr = cast([ap - 4], felt*);
-            let pedersen_ptr = cast([ap - 3], HashBuiltin*);
-            let range_check_ptr = [ap - 2];
-            let caller_starknet_address = [ap - 1];
-
-            let (retdata_len, retdata, success) = IAccount.execute_starknet_call(
-                caller_starknet_address, to_starknet_address, starknet_selector, data_len, data
-            );
-            if (success == FALSE) {
-                // skip formatting to bytes32 array and return revert reason directly
-                return (retdata_len, retdata, CAIRO_PRECOMPILE_GAS, TRUE);
-            }
+        return Internals.execute_cairo_call(
+            caller_address,
+            call_selector,
+            to_starknet_address,
+            starknet_selector,
+            data_len,
+            data,
+            skip_returndata=FALSE,
+        );
+    }
 
-            let (output) = alloc();
-            let output_len = retdata_len * 32;
-            Helpers.felt_array_to_bytes32_array(retdata_len, retdata, output);
-            return (output_len, output, CAIRO_PRECOMPILE_GAS, FALSE);
+    // @notice Executes a batch of call to cairo contracts.
+    // @dev Cannot be called with CALLCODE / DELEGATECALL
+    // @param input_len The length of the input in bytes.
+    // @param input The input data.
+    // @param caller_address The address of the caller of the precompile.
+    func multicall_cairo_precompile{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(input_len: felt, input: felt*, caller_address: felt) -> (
+        output_len: felt, output: felt*, gas_used: felt, reverted: felt
+    ) {
+        alloc_locals;
+
+        // Input must be at least 8 bytes long.
+        let is_input_invalid = is_nn(8 - input_len);
+        if (is_input_invalid != 0) {
+            let (revert_reason_len, revert_reason) = Errors.outOfBoundsRead();
+            return (
+                revert_reason_len, revert_reason, CAIRO_PRECOMPILE_GAS, Errors.EXCEPTIONAL_HALT
+            );
         }
 
-        let (revert_reason_len, revert_reason) = Errors.invalidCairoSelector();
-        return (revert_reason_len, revert_reason, CAIRO_PRECOMPILE_GAS, TRUE);
+        // Input is formatted as:
+        // [call_selector: bytes4][number_of_calls: bytes4]
+        // [starknet_address_1: bytes32][starknet_selector_1:bytes32][data_offset_1: bytes32][data_len_1: bytes32][data_1: bytes[]]...
```

Comment:
isn't this the cairo0 formatting? cairo calls are 
- `to`
- `selector`
- `calldata_len`
- `calldata*`

let's just adopt this convention

Reply:
this is the EVM encoding of a cairo call